server/src/main/java/com/decathlon/ara/cartography/SquashExporter.java

```diff
@@ -27,8 +54,117 @@ public String getFormat() {
+                String criticity = CRITICALITY_MAPPING.get(functionality.getSeverity());
+                String funcName = functionality.getName();
+                String path = "/" + projectNameInSquash + this.getFunctionalityPath(functionality, functionalities);
+                this.addRowToSheet(sheet, idx++, projectNameInSquash,
+                        path + "/" + funcName, functionality.getComment(), criticity,
+                        importDate, userWhichMakesTheImport);
+            }
+        }
+
+        try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream()) {
+            workbook.write(outputStream);
+            workbook.close();
+            return outputStream.toByteArray();
+        } catch (IOException ex) {
+            log.error("Unable to write the Squash export : {}", ex.getMessage());
+            log.debug("Full exception of the Squash export failure", ex);
+        }
         return new byte[0];
     }
+
+    @Override
+    public List<ExportField> listRequiredFields() {
+        List<ExportField> fields = new ArrayList<>();
+        fields.add(new ExportField(PROJECT_NAME, "Targeted project's name", "string",
+                "The name of targeted project in SquashTM. Note that the project must be created before doing the import."));
+        fields.add(new ExportField(USER, "Targeted User's name", "string",
+                "The name of the user whom will act as the creator of the requirements in SquashTM. Note that the user must exist before doing the import."));
+        return fields;
+    }
+
+    private Row createHeaderRow(Sheet sheet) {
+        Row header = sheet.createRow(0);
+        int rowIdx = 0;
+        this.addCellToRow(header, rowIdx++, "ACTION");
+        this.addCellToRow(header, rowIdx++, "PROJECT_ID");
+        this.addCellToRow(header, rowIdx++, "PROJECT_NAME");
+        this.addCellToRow(header, rowIdx++, "REQ_PATH");
+        this.addCellToRow(header, rowIdx++, "REQ_NUM");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_NUM");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_NAME");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_CRITICALITY");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_CATEGORY");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_STATUS");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_DESCRIPTION");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_#_TC");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_#_ATTACHEMENT");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_CREATED_ON");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_CREATED_BY");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_LAST_MODIFIED_ON");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_LAST_MODIFIED_BY");
+        this.addCellToRow(header, rowIdx++, "REQ_VERSION_MILESTONE");
+        this.addCellToRow(header, rowIdx, "REQ_VERSION_CUF_<code du cuf>");
```

Comment:
What about using an array storing all of your attribute (e.g. `"ACTION"`, `"PROJECT_ID"`, etc.), then looping through it meanwhile adding your cells ?

Reply:
Ok for the headers :)